### PR TITLE
backport: merge bitcoin#10508, #18051, #19683, #19688, #19761, #19015, #19916, #20527, #21505, #21629, partial bitcoin#18307, #21205 (build backports)

### PR DESCRIPTION
**Pull request by kwvg**

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for merging via merge commit

File: depends/packages/zeromq.mk [hunk 1]
```diff
@@ -34,6 +36,5 @@ define $(package)_stage_cmds
 endef
 
 define $(package)_postprocess_cmds
-  sed -i.old "s/ -lstdc++//" lib/pkgconfig/libzmq.pc && \
```
[the remark this answers is not in the document]
Comment (kwvg) [reply depends/packages/zeromq.mk:37]: * bitcoin#19761 (the backport in question) was merged in Aug 2020
* The `libxcb` `sed` replacement with a patch is done in bitcoin#25738, which was merged in Jul 2022 and should therefore have no effect on the above PR's backport status
* The `ret_sig` patch is no longer applicable as the patch itself is dropped in [bitcoin#19764](https://github.com/bitcoin/bitcoin/pull/19764/files#diff-1b9f971ca3cf2a0e60e7529f743f8adc7e8bac395f89c0e2c4db735bfb06aebd), which was backported in [dash#4356](https://github.com/dashpay/dash/pull/4356/commits/6a367fc1987d449d72a9561aee8e980a544dc9e9), meaning the appropriate action would be to drop the `sed` command altogether. (I had explained this circumstance in the commit comment itself)
* The `unary_function` patch is from [dash#5610](https://github.com/dashpay/dash/pull/5610/files#diff-2d320b9a315a1970ad66d8ec4a5d59323ecf35edd3c893d67e7e9b0bf5ee6c69L38-R46) and is a Dash-ism. Converting that into a patch is outside the scope of the backport and shouldn't affect its merge status.
* The `arm-linux-gnueabi` `sed` replacement survives all the way till the last v23 release of Bitcoin ([source](https://github.com/bitcoin/bitcoin/blob/v23.2/depends/packages/qt.mk#L248)) and is replaced in bitcoin#25838 (merged Aug 2022) with a larger set of `sed` commands. Those commands, as of this writing, are present in `master` ([source](https://github.com/bitcoin/bitcoin/blob/063a8b83875997068b3eb506b5f30f2691d18052/depends/packages/qt.mk#L261-L266))
* The `QMAKE_C`(`C`/`XX`) replacements have been untouched since when it was first introduced in bitcoin#21376 (merged Mar, 2021) and bitcoin#25719 (merged Aug 2022) respectively and continue to remain in `master` as of this writing ([source](https://github.com/bitcoin/bitcoin/blob/063a8b83875997068b3eb506b5f30f2691d18052/depends/packages/qt.mk#L270-L271))

**TL;DR:** The patches as introduced in bitcoin#19761, the PR in scope, were properly backported and in fact, one patch needs to be dropped due to inapplicability. `sed` usage hasn't gone anywhere and will be here to stay in the foreseeable future. Backports have been selected on the suggestion that the current and immediate next base version of upstream changes be pulled downstream and so, OOO backports have not been strongly favored.

**TL;TL;DR:** Dropped `ret_sig` replacement and comment, we're on Boost 1.77 now! (see c588ea52a89697c7f81e8090832214af616ea106)

**Review by knst — DISMISSED** [review]
utACK

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — DISMISSED** [review]
utACK

File: doc/build-osx.md [hunk 2]
```diff
@@ -19,13 +19,27 @@ Then install [Homebrew](https://brew.sh).
...
-brew install automake berkeley-db4 libtool boost gmp miniupnpc pkg-config python qt libevent libnatpmp qrencode sqlite
+brew install automake libtool boost gmp miniupnpc pkg-config python qt libevent libnatpmp qrencode
 ```
 
 If you run into issues, check [Homebrew's troubleshooting page](https://docs.brew.sh/Troubleshooting).
 See [dependencies.md](dependencies.md) for a complete overview.
 
-## Berkeley DB
+The wallet support requires one or both of the dependencies ([*SQLite*](#sqlite) and [*Berkeley DB*](#berkeley-db)) in the sections below.
+To build Bitcoin Core without wallet, see [*Disable-wallet mode*](#disable-wallet-mode).
```
[the remark this answers is not in the document]
Comment (kwvg) [reply doc/build-osx.md]: Resolved in b6b0a3529adb476b2c158523e797876d4d56941d

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK, diff is as expected
